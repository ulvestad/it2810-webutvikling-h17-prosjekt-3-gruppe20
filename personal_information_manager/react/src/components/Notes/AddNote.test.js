import React from 'react';
import sinon from 'sinon';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { expect } from 'chai';

import AddNote from './AddNote';

const jestExpect = global.expect;

it('renders without crashing', () => {
  shallow(<AddNote />);
});

it('renders correctly without props', () => {
  const tree = renderer.create(<AddNote />).toJSON();
  jestExpect(tree).toMatchSnapshot();
});

it('renders correctly with props', () => {
  const handleClick = () => {console.log('clicked')};

  const addNoteComponent =
      <AddNote handleClick={handleClick} />;

  const tree = renderer.create(addNoteComponent).toJSON();

  jestExpect(tree).toMatchSnapshot();
});

it('calls handler when clicked', () => {
  const handler = sinon.spy();
  const wrapper = shallow(<AddNote handleClick={handler} />);

  wrapper.find('.panel').simulate('click');
  expect(handler).to.have.property('callCount', 1);
});
