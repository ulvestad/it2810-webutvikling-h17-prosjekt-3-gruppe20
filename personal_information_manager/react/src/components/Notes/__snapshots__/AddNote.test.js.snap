// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with props 1`] = `
<div
  className="panel panel-default note add-note"
  onClick={[Function]}
>
  <div
    className="panel-body"
  >
    <i
      aria-hidden="true"
      className="fa fa-plus add"
    />
  </div>
</div>
`;

exports[`renders correctly without props 1`] = `
<div
  className="panel panel-default note add-note"
  onClick={undefined}
>
  <div
    className="panel-body"
  >
    <i
      aria-hidden="true"
      className="fa fa-plus add"
    />
  </div>
</div>
`;
